WooCommerce/Classes/ViewRelated/Products/Add Product/AddProductWithAI/StartingInfo/PromptProgressBar/ProductCreationAIPromptProgressBarViewModel.swift

```diff
@@ -0,0 +1,133 @@
+    @Published private(set) var status: ProgressStatus = .start
+    @Published var text: String = "" {
+        didSet {
+            updateStatusBasedOnText(text)
+        }
+    }
+
+    // Method for updating the text
+    func updateText(to newText: String) {
+        text = newText
+    }
+
+    // Method for updating the status based on text
+    private func updateStatusBasedOnText(_ text: String) {
+        let wordCount = text.split { $0 == " " || $0.isNewline }.count
+
+        switch wordCount {
+        case 0:
+            status = .start
+        case 1...10:
+            status = .inProgress
+        case 11...19:
+            status = .halfway
+        case 20...30:
+            status = .almostDone
+        default:
+            status = .completed
+        }
+    }
+
+    enum ProgressStatus: Int, CaseIterable {
+        case start, inProgress, halfway, almostDone, completed
+
+        var progress: Float {
+            switch self {
+            case .start:
+                return 0.03
+            case .inProgress:
+                return 0.2
+            case .halfway:
+                return 0.4
+            case .almostDone:
+                return 0.7
+            case .completed:
+                return 0.9
+            }
+        }
+
+        var color: Color {
+            switch self {
+            case .start:
+                return Color.init(uiColor: .gray(.shade50))
+            case .inProgress:
+                return Color.init(uiColor: .withColorStudio(.red, shade: .shade50))
+            case .halfway:
+                return Color.init(uiColor: .withColorStudio(.orange, shade: .shade50))
+            case .almostDone:
+                return Color.init(uiColor: .withColorStudio(.yellow, shade: .shade50))
+            case .completed:
+                return Color.init(uiColor: .withColorStudio(.green, shade: .shade50))
+            }
+        }
+
+        var mainDescription: String {
+            switch self {
+            case .start:
+                return ""
+            case .inProgress:
+                return NSLocalizedString(
+                    "productCreationAIPromptProgressBar.main.inProgress",
+                    value: "Add more details. ",
+                    comment: "State when more details need to be added for the main prompt description suggestion in product creation with AI."
+                )
+            case .halfway:
+                return NSLocalizedString(
+                    "productCreationAIPromptProgressBar.main.halfway",
+                    value: "Getting better. ",
+                    comment: "State when the prompt description is improving for the main prompt description suggestion in product creation with AI."
+                )
+            case .almostDone:
+                return NSLocalizedString(
+                    "productCreationAIPromptProgressBar.main.almostDone",
+                    value: "Great prompt! ",
```

Comment:
I think this should be "Almost there.". 

<img width="266" alt="image" src="https://github.com/woocommerce/woocommerce-ios/assets/524475/289e311b-ebdb-4401-a6f0-f65edd8f0193">